bakery/service/update_generation.go

```diff
@@ -0,0 +1,187 @@
+package service
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strings"
+)
+
+func GenerateImagesTree(previousImageName string, recursive, skipExistingDirectories bool, nameReplacements []string) error {
+	previousDockerImage := hierarchy.GetImageByName(previousImageName)
+	if previousDockerImage == nil {
+		return fmt.Errorf("unable to find image %s in the analyzed structure (is invocation directory correct?)", previousImageName)
+	}
+
+	replacer := createNewNameReplacer(nameReplacements)
+	return generateAncestors(previousImageName, dependencies, replacer, recursive, skipExistingDirectories)
+}
+
+func createNewNameReplacer(nameReplacements []string) nameReplacer {
+	replacer := newNameReplacer()
+	for _, replacement := range nameReplacements {
+		keyValuePair := strings.Split(replacement, propertyKeyValueSeparator)
+		if len(keyValuePair) != 2 {
+			fmt.Printf("Unable to parse provided replacement: %s - expecting key and value to be separated with '%s'", replacement, propertyKeyValueSeparator)
+			continue
+		}
+		replacer = replacer.add(keyValuePair[0], keyValuePair[1])
+	}
+	return replacer
+}
+
+func generateAncestors(previousImageName string, dependenciesMap map[string][]*DockerImage, replacer nameReplacer, recursive bool, skipExistingDirectories bool) error {
+	ancestorsToGenerate, err := getChildrenToUpdate(previousImageName, dependenciesMap, replacer, recursive)
+	fmt.Println("Found ancestors to recreate:")
```

Comment:
I'd suggest extracting `printAncestors` function just for a sake of better readability